protocol-contracts/staking/test/OperatorRewarder.test.ts

```diff
@@ -241,6 +245,48 @@ describe('OperatorRewarder', function () {
       // Historical reward: (1+10) (seconds) * 0.5 (reward rate) = 10.5
       await expect(this.mock.historicalReward()).to.eventually.eq(ethers.parseEther('10.5'));
     });
+
+    it('should claim rewards with authorized claimer', async function () {
+      await this.operatorStaking.connect(this.delegator1).deposit(ethers.parseEther('1'), this.delegator1);
+      await this.mock.connect(this.delegator1).setClaimer(this.claimer);
+      await expect(this.mock.claimer(this.delegator1)).to.eventually.eq(this.claimer);
+
+      await timeIncreaseNoMine(10);
+      await this.protocolStaking.connect(this.admin).setRewardRate(ethers.parseEther('0'));
+
+      await expect(this.mock.connect(this.claimer).claimRewards(this.delegator1))
+        .to.emit(this.token, 'Transfer')
+        .withArgs(this.mock, this.delegator1, ethers.parseEther('5.5'));
+    });
+
+    it('should claim rewards with delegator first and then authorized claimer', async function () {
+      await this.operatorStaking.connect(this.delegator1).deposit(ethers.parseEther('1'), this.delegator1);
+      await expect(this.mock.claimer(this.delegator1)).to.eventually.eq(this.delegator1);
+
+      await timeIncreaseNoMine(10);
+      await this.protocolStaking.connect(this.admin).setRewardRate(ethers.parseEther('0'));
+
+      await expect(this.mock.connect(this.delegator1).claimRewards(this.delegator1))
+        .to.emit(this.token, 'Transfer')
+        .withArgs(this.mock, this.delegator1, ethers.parseEther('5'));
+
+      await this.protocolStaking.connect(this.admin).setRewardRate(ethers.parseEther('0.5'));
+      await timeIncreaseNoMine(10);
+      await this.protocolStaking.connect(this.admin).setRewardRate(ethers.parseEther('0'));
+
+      await this.mock.connect(this.delegator1).setClaimer(this.claimer);
+
+      await expect(this.mock.connect(this.claimer).claimRewards(this.delegator1))
+        .to.emit(this.token, 'Transfer')
+        .withArgs(this.mock, this.delegator1, ethers.parseEther('5'));
+    });
+
+    it('should not claim rewards if not authorized to claim rewards', async function () {
+      await expect(this.mock.claimer(this.anyone)).to.not.eventually.eq(this.claimer);
```

Comment:
```suggestion
      await expect(this.mock.claimer(this.delegator1)).to.not.eventually.eq(this.claimer);
```
This is what you meant?

Reply:
yes good catch !